codegen/type_converter_test.go

```diff
@@ -966,3 +1026,542 @@ func TestConvertWithBadKeyMapOfString(t *testing.T) {
 		err.Error(),
 	)
 }
+
+func TestConverterMap(t *testing.T) {
+	fieldMap := make(map[string]codegen.FieldMapperEntry)
+	fieldMap["One"] = codegen.FieldMapperEntry{
+		QualifiedName: "Two",
+		Override:      true,
+	}
+	fieldMap["Two"] = codegen.FieldMapperEntry{
+		QualifiedName: "One",
+		Override:      true,
+	}
+
+	lines, err := convertTypes(
+		"Foo", "Bar",
+		`struct Foo {
+			1: required bool one
+			2: required bool two
+		}
+
+		struct Bar {
+			1: required bool one
+			2: required bool two
+		}`,
+		nil,
+		fieldMap,
+	)
+
+	assert.NoError(t, err)
+	assert.Equal(t, trim(`
+		out.One = bool(in.Two)
+		out.Two = bool(in.One)
+	`), lines)
+}
+
+func TestConverterMapNewField(t *testing.T) {
+	fieldMap := make(map[string]codegen.FieldMapperEntry)
+	fieldMap["Two"] = codegen.FieldMapperEntry{
+		QualifiedName: "One",
+		Override:      true,
+	}
+
+	lines, err := convertTypes(
+		"Foo", "Bar",
+		`struct Foo {
+			1: required bool one
+		}
+
+		struct Bar {
+			1: required bool one
+			2: required bool two
+		}`,
+		nil,
+		fieldMap,
+	)
+
+	assert.NoError(t, err)
+	assert.Equal(t, trim(`
+		out.One = bool(in.One)
+		out.Two = bool(in.One)
+	`), lines)
+}
+
+func TestConverterMapOptionalNoOverride(t *testing.T) {
+	fieldMap := make(map[string]codegen.FieldMapperEntry)
+	fieldMap["One"] = codegen.FieldMapperEntry{
+		QualifiedName: "Two",
+		Override:      false,
+	}
+
+	lines, err := convertTypes(
+		"Foo", "Bar",
+		`struct Foo {
+			1: optional bool one
+			2: optional bool two
+		}
+
+		struct Bar {
+			1: optional bool one
+			2: optional bool two
+		}`,
+		nil,
+		fieldMap,
+	)
+
+	assert.NoError(t, err)
+	assert.Equal(t, trim(`
+		out.One = (*bool)(in.Two)
+		if in.One != nil {
+			out.One = (*bool)(in.One)
+		}
+		out.Two = (*bool)(in.Two)
+	`), lines)
+}
+
+func TestConverterMapOverrideOptional(t *testing.T) {
+	fieldMap := make(map[string]codegen.FieldMapperEntry)
+	fieldMap["One"] = codegen.FieldMapperEntry{
+		QualifiedName: "Two",
+		Override:      true,
+	} // Map from required filed, unconditional assignment
+	fieldMap["Two"] = codegen.FieldMapperEntry{
```

Comment:
It's confusing to set "override: true" but still have the field come from `in.Two` instead of `in.One`

Maybe we should make both fields required if override is true and fail the code generation if one of them is optional.